custom_components/healthchecksio/__init__.py

```diff
@@ -74,12 +75,31 @@ async def async_setup_entry(hass, config_entry):
 
     # Add binary_sensor
     hass.async_add_job(
-        hass.config_entries.async_forward_entry_setup(config_entry, "binary_sensor")
+        hass.config_entries.async_forward_entry_setup(
+            config_entry, Platform.BINARY_SENSOR
+        )
     )
 
     return True
 
 
+async def async_unload_entry(
+    hass: core.HomeAssistant, config_entry: config_entries.ConfigEntry
+) -> bool:
+    """Unload a config entry."""
+
+    unload_ok = await hass.config_entries.async_forward_entry_unload(
+        config_entry, Platform.BINARY_SENSOR
+    )
+    if unload_ok:
+        hass.data[DOMAIN_DATA].pop("client", None)
+        hass.data[DOMAIN_DATA].pop("data", None)
```

Comment:
Why not just evict all DOMAIN_DATA here?
`hass.data.pop(DOMAIN_DATA, None)`

Reply:
Makes sense.